tuf/api/metadata.py

```diff
@@ -455,13 +455,16 @@ def to_dict(self) -> JsonDict:
 
 
     # Modification.
-    def update(self, version: int, length: int, hashes: JsonDict) -> None:
+    def update(self, version: int, length: Optional[int] = None,
+            hashes: Optional[JsonDict] = None) -> None:
```

Comment:
I'm wondering if `JsonDict` is the right type here, and elsewhere in the API? I wonder whether in the interest of not encoding our wireline format in the API we should have a more specialised type for what the specification calls Objects?

Maybe the solution is as simple as renaming `JsonDict` to something else?

I'm happy for this to be addressed in a future change, but wanted to hear what others think.

Reply:
Agreed. Maybe `StrDict` is less biased. Or we just use the aliased value, i.e. `Dict[str, Any]`, which is not that much longer. Maybe this is something we could revisit with #1140?

Reply:
I don't have a good name suggestion, so would lean towards the explicit aliased value of `Dict[str, Any]` to start with. I'm more than happy to revisit in a future PR.

Reply:
I agree that something more specific is needed.
`StrDict` sounds good. I am not sure `Dict[str, Any]` is what we need, especially that values could be `Any` when they
should be strings. Maybe like `Dict[str, str]` is better?